src/cmd/linuxkit/push_packet.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"compress/gzip"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	packetDefaultArch       = "x86_64"
+	packetDefaultDecompress = false
+)
+
+func init() {
+	if runtime.GOARCH == "arm64" {
+		packetDefaultArch = "aarch64"
+		// decompress on arm64. iPXE/kernel does not
+		// seem to grok compressed kernels/initrds.
+		packetDefaultDecompress = true
+	}
+}
+
+// Process the run arguments and execute run
+func pushPacket(args []string) {
+	flags := flag.NewFlagSet("packet", flag.ExitOnError)
+	invoked := filepath.Base(os.Args[0])
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s push packet [options] [name]\n\n", invoked)
+		fmt.Printf("Options:\n\n")
+		flags.PrintDefaults()
+	}
+	baseURLFlag := flags.String("base-url", "", "Base URL that the kernel, initrd and iPXE script are served from (or "+packetBaseURL+")")
+	nameFlag := flags.String("img-name", "", "Overrides the prefix used to identify the files. Defaults to [name] (or "+packetNameVar+")")
+	archFlag := flags.String("arch", packetDefaultArch, "Image architecture (x86_64 or aarch64)")
+	decompressFlag := flags.Bool("decompress", packetDefaultDecompress, "Decompress kernel/initrd before pushing")
+	dstFlag := flags.String("destination", "", "URL where to push the image to. Currently only 'file' is supported as a scheme (which is also the default if omitted)")
```

Comment:
The default given here is `""` and later on there is a `log.Fatal` inside an `if *dstFlag == ""` with nothing to set a default of `file` that I see in between.

Reply:
Oh, I see what you mean now, i misread.... only the scheme is defaulted, but you do need a path. gotcha.